pkgs/shells/marcel/default.nix

```
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, bash
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "marcel";
+  version = "0.17.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "geophile";
+    repo = "marcel";
+    rev = "v${version}";
+    hash = "sha256-JMDZiZcsSXWxyx2ZJMihlzl62FPk+V8HWejQwZObpUs=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    dill
+    psutil
+    bash
+  ];
+
+  prePatch = ''
+    substituteInPlace marcel/op/bash.py experiments/signal_propagation.py marcel/op/edit.py experiments/cp_behavior.py experiments/popen_stdout_stderr.py \
+        --replace '/bin/bash' ${bash}/bin/bash
+  '';
+
+  doCheck = false; # cause the tests provided by marcel will read/write $HOME/.local/share/marcel and /tmp and use sudo
+
+  postFixup = ''
+    substituteInPlace $out/bin/marcel \
+    --replace python3 ${python3}/bin/python3
+    wrapProgram $out/bin/marcel \
+    --prefix PYTHONPATH : "$program_PYTHONPATH"
+    '';
```

```suggestion
  postFixup = ''
    substituteInPlace $out/bin/marcel \
        --replace python3 ${python3}/bin/python3
    wrapProgram $out/bin/marcel \
        --prefix PATH : "$program_PATH:${bash}/bin" \
        --prefix PYTHONPATH : "$program_PYTHONPATH"
    '';
```
It's recommended to use indentation (4 spaces) in multiline shell commands.

Also, please add `$program_PATH` and `${bash}/bin` to `$PATH`.
- `$program_PATH` is just `$out/bin`, but this is variable is what the original python wrapper uses, so I'll recommend this. The reason this is used is for it to be able to find `farcel.py` (for running sudo commands)
- `${bash}/bin` is needed in case some weird person somehow doesn't have bash installed. (The program assumes that `bash` is available)